src/Compiler/Microsoft.AspNetCore.Razor.Language/src/RazorCodeDocumentExtensions.cs

```diff
@@ -550,7 +561,16 @@ public override void VisitRazorDirective(RazorDirectiveSyntax node)
                 if (directiveContent != null && directiveContent.StartsWith(NamespaceDirective.Directive.Directive, StringComparison.Ordinal))
                 {
                     LastNamespaceContent = directiveContent.Substring(NamespaceDirective.Directive.Directive.Length).Trim();
-                    LastNamespaceLocation = node.GetSourceSpan(_source);
+                    var location = node.GetSourceSpan(_source);
+
+                    var offset = 1 + NamespaceDirective.Directive.Directive.Length + 1; // @ + namespace + " "
+                    LastNamespaceLocation = new SourceSpan(location.FilePath,
```

Comment:
I hate everything about this. We need to change how all this stuff is parsed to just put the actual things into the tree, rather than trying to reconstruct this later.

Reply:
100% agree. I looked at doing that but it's not trivial and I don't want to block fuse progress on  it.